tactics/tactics.ml

```diff
@@ -4250,10 +4250,11 @@ let find_induction_type isrec elim hyp0 gl =
     match elim with
     | None ->
        let sort = Tacticals.New.elimination_sort_of_goal gl in
-       let sigma, (elimc,elimt),_ = guess_elim isrec false sort hyp0 gl in
-       let scheme = compute_elim_sig sigma ~elimc elimt in
-       (* We drop the scheme waiting to know if it is dependent *)
-       sigma, scheme, ElimOver (isrec,hyp0)
+       let sigma', (elimc,elimt),_ = guess_elim isrec false sort hyp0 gl in
+       let scheme = compute_elim_sig sigma' ~elimc elimt in
+       (* We drop the scheme waiting to know if it is dependent, this
+          needs no update to sigma at this point. *)
+       Tacmach.New.project gl, scheme, ElimOver (isrec,hyp0)
```

Comment:
This is correct, but the way to code is written doesn't make it obvious that the newer evarmap is never going to be referred to in the return branches. I wouldn't mind marking this a in need of a TODO.

Reply:
Why not, we could also just return the indref and nparams instead of the scheme which might otherwise depend on sigma'

Reply:
Done